### ISSUE TITLE
[Epic] Honey SQL 2 Migration, Continued [47-49]

### ISSUE THREAD

**Issue by camsaul** [metabase/metabase#28684]
Splitting off the 47+ work from #28009

```[tasklist]
### In 47 or 48:
- [ ] #28156
- [ ] #28161
- [ ] #28169
- [ ] Make Honey SQL 2 the default `honey-sql-version`. Driver authors can still use Honey SQL 1, but will have to implement this method and return `1` instead of `2` (instead of vice versa). Update driver changelog to mention this
```

Once the 46 stuff is done we can probably move this into a separate issue
```[tasklist]
### In 49:
- [ ] #28165
- [ ] https://github.com/metabase/metabase/issues/28396
```
